### ISSUE TITLE
FTBFS: test failure in t/signer.t with CryptX 0.081

### ISSUE THREAD

**Issue by gregoa** [fastmail/mail-dkim#33]
We have the following bug reported to the Debian package of
Mail-DKIM, c.f. https://bugs.debian.org/1082542

It doesn't seem to be a bug in the packaging, so you may want to take
a look. Thanks!

------8<-----------8<-----------8<-----------8<-----------8<-----

```
Source: libmail-dkim-perl
Version: 1.20240827-1
Severity: serious
Tags: upstream ftbfs
Justification: fails to build from source

After updating CryptX to 0.081 one test fails:

t/signer.t .. 
1..35
ok 1 - new() works
ok 2 - signature() works
ok 3 - got expected signature value
ok 4 - new() works
ok 5 - signature() works
ok 6 - new() works
ok 7 - signature() works
ok 8 - subject was signed
ok 9 - from was signed
ok 10 - received was excluded
ok 11 - comments was excluded
ok 12 - got expected identity value
ok 13 - found timestamp value
ok 14 - new() works
ok 15 - signature() works
ok 16 - subject was over signed
ok 17 - x-test was all signed
ok 18 - x-tests was signed
ok 19 - x-tests was signed only once
ok 20 - from was signed
ok 21 - date was excluded
ok 22 - comments was excluded
ok 23 - got expected identity value
ok 24 - found timestamp value
ok 25 - new() with bogus key file dies as expected
ok 26 - new() with bogus key file dies as expected
ok 27 - signature() works
ok 28 - signature() works
ok 29 - signature() works
ok 30 - subject was signed
ok 31 - from was signed
ok 32 - new() works
Ed25519 failed: FATAL: invalid or unsupported Ed25519 key format at /build/libmail-dkim-perl-1.20240827/blib/lib/Mail/DKIM/PrivateKey.pm line 106.
	...propagated at /build/libmail-dkim-perl-1.20240827/blib/lib/Mail/DKIM/PrivateKey.pm line 172.
Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 3/35 subtests 


Cheers,
gregor


```
------8<-----------8<-----------8<-----------8<-----------8<-----

Thanks for considering,
  gregor herrmann,
  Debian Perl Group

**Comment by notbrokenjemma** [comment]
Fixed by #34

**Comment by notbrokenjemma** [comment]
Thanks for the report @gregoa 
This has been fixed in 1.20240923 and released to CPAN

**Comment by gregoa** [comment]
On Sun, 22 Sep 2024 21:59:20 -0700, Marc Bradshaw wrote:

> Thanks for the report @gregoa 
> This has been fixed in 1.20240923 and released to CPAN

Thanks for the super-fast fix!

Uploaded to Debian/unstable.


Cheers,
gregor 

-- 
 .''`.  https://info.comodo.priv.at -- Debian Developer https://www.debian.org
 : :' : OpenPGP fingerprint D1E1 316E 93A7 60A8 104D  85FA BB3A 6801 8649 AA06
 `. `'  Member VIBE!AT & SPI Inc. -- Supporter Free Software Foundation Europe
   `-
